include/unifex/finally.hpp

```diff
@@ -700,39 +701,41 @@ namespace unifex
         : source_(static_cast<SourceSender2&&>(source))
         , completion_(static_cast<CompletionSender2&&>(completion)) {}
 
-      // TODO: Also constrain this method to check that the CompletionSender
+      // TODO: Also constrain these methods to check that the CompletionSender
       // is connectable to any of the instantiations of done/value/error_receiver
       // that could be created for each of the results that SourceSender might
       // complete with. For now we just check done_receiver as an approximation.
+
       template <
           typename Receiver,
           typename CPO,
           typename S,
           std::enable_if_t<
               std::conjunction_v<
                 std::is_same<CPO, tag_t<connect>>,
-                std::is_same<S, sender>,
+                std::is_same<std::remove_cvref_t<S>, sender>,
+                std::negation<std::is_lvalue_reference<S>>,
                 is_connectable<
                   SourceSender,
                   receiver<
                       SourceSender,
                       CompletionSender,
-                      Receiver>>,
+                      std::remove_cvref_t<Receiver>>>,
                 is_connectable<
                   CompletionSender,
                   done_receiver<
                       SourceSender,
                       CompletionSender,
-                      Receiver>>>,
-              int> = 0>
+                      std::remove_cvref_t<Receiver>>>,
+                std::true_type>,
```

Comment:
Why was it necessary to add `std::true_type` here?
I thought std::conjunction would have had the same behaviour without it?

Reply:
It allowed me to comment out full lines easily. removed.